src/main/java/com/contentgrid/gateway/security/oidc/OidcClientConfiguration.java

```diff
@@ -49,22 +50,35 @@ ReactiveClientRegistrationIdResolver clientRegistrationIdResolver(
                     .map("client-%s"::formatted);
         }
 
+        @Bean
+        ReactiveClientRegistrationResolver oAuth2ClientApplicationConfigurationMapper(
+                ReactiveClientRegistrationIdResolver registrationIdResolver) {
+            return new OAuth2ClientApplicationConfigurationMapper(registrationIdResolver);
+        }
+
         @Bean
         ReactiveClientRegistrationRepository clientRegistrationRepository(
                 ReactiveClientRegistrationIdResolver registrationIdResolver,
+                ReactiveClientRegistrationResolver clientRegistrationResolver,
                 ApplicationConfigurationRepository applicationConfigurationRepository,
                 OAuth2ClientProperties properties) {
 
-            var mapper = new OAuth2ClientApplicationConfigurationMapper(registrationIdResolver);
             return new DynamicReactiveClientRegistrationRepository(applicationConfigurationRepository
                     .observe()
-                    .flatMap(update -> registrationIdResolver
-                            .resolveRegistrationId(update.getKey())
```

Comment:
When update type is `CLEAR`, then `update.getKey())` returns `null`, causing a NPE